### ISSUE TITLE
Out-of-bound write in CUDA Basic_INIT_VIEW1D_OFFSET

### ISSUE THREAD

**Issue by abbotts** [LLNL/RAJAPerf#86]
I'm seeing an out-of-bound write in the `Basic_INIT_VIEW1D_OFFSET` test built for V100 with GCC 8.1.0 as the host compiler on RHEL 8.2. From cuda-memcheck on a Debug build:

```
CUDAassert: unspecified launch failure /home/users/sabbott/cuda_tests/RAJAPerf/tpl/RAJA/include/RAJA/policy/cuda/MemUtils_CUDA.hpp 210
terminate called after throwing an instance of 'std::runtime_error'
  what():  CUDAassert
========= CUDA-MEMCHECK
========= Invalid __global__ write of size 8
=========     at 0x00000140 in _ZN4RAJA6policy4cuda4impl18forall_cuda_kernelILm256ENS_9Iterators16numeric_iteratorIllPlEEZN8rajaperf5basic18INIT_VIEW1D_OFFSET14runCudaVariantENS8_9VariantIDEEUllE_lEEvT1_T0_T2_
=========     by thread (0,0,0) in block (0,0,0)
=========     Address 0x15552c5ffff8 is out of bounds
=========     Device Frame:_ZN4RAJA6policy4cuda4impl18forall_cuda_kernelILm256ENS_9Iterators16numeric_iteratorIllPlEEZN8rajaperf5basic18INIT_VIEW1D_OFFSET14runCudaVariantENS8_9VariantIDEEUllE_lEEvT1_T0_T2_ (_ZN4RAJA6policy4cuda4impl18forall_cuda_kernelILm256ENS_9Iterators16numeric_iteratorIllPlEEZN8rajaperf5basic18INIT_VIEW1D_OFFSET14runCudaVariantENS8_9VariantIDEEUllE_lEEvT1_T0_T2_ : 0x140)
=========     Saved host backtrace up to driver entry point at kernel launch time
=========     Host Frame:/cm/local/apps/cuda/libs/current/lib64/libcuda.so.1 (cuLaunchKernel + 0x346) [0x2c4fd6]
=========     Host Frame:raja-perf.exe [0x155449]
=========     Host Frame:raja-perf.exe [0x1554d7]
=========     Host Frame:raja-perf.exe [0x18b835]
=========     Host Frame:raja-perf.exe [0x57c7e]
=========     Host Frame:raja-perf.exe [0x15492]
=========     Host Frame:raja-perf.exe [0x152ab]
=========     Host Frame:raja-perf.exe [0x89ad]
=========     Host Frame:raja-perf.exe [0x7159]
=========     Host Frame:/lib64/libc.so.6 (__libc_start_main + 0xf3) [0x236a3]
=========     Host Frame:raja-perf.exe [0x6fce]
=========
========= Program hit cudaErrorLaunchFailure (error 719) due to "unspecified launch failure" on CUDA API call to cudaLaunchKernel.
=========     Saved host backtrace up to driver entry point at error
=========     Host Frame:/cm/local/apps/cuda/libs/current/lib64/libcuda.so.1 [0x3da513]
=========     Host Frame:raja-perf.exe [0x18b875]
=========     Host Frame:raja-perf.exe [0x57c7e]
=========     Host Frame:raja-perf.exe [0x15492]
=========     Host Frame:raja-perf.exe [0x152ab]
=========     Host Frame:raja-perf.exe [0x89ad]
=========     Host Frame:raja-perf.exe [0x7159]
=========     Host Frame:/lib64/libc.so.6 (__libc_start_main + 0xf3) [0x236a3]
=========     Host Frame:raja-perf.exe [0x6fce]
=========
========= Error: process didn't terminate successfully
========= No CUDA-MEMCHECK results found
```
The kernel demangles to `void RAJA::policy::cuda::impl::forall_cuda_kernel<256ul, RAJA::Iterators::numeric_iterator<long, long, long*>, rajaperf::basic::INIT_VIEW1D_OFFSET::runCudaVariant(rajaperf::VariantID)::{lambda(long)#1}, long>(rajaperf::basic::INIT_VIEW1D_OFFSET::runCudaVariant(rajaperf::VariantID)::{lambda(long)#1}, RAJA::Iterators::numeric_iterator<long, long, long*>, long)`

I configured with
```
cmake    -DCMAKE_BUILD_TYPE=Debug     -DENABLE_OPENMP=Off     -DENABLE_CUDA=On  -DCUDA_ARCH=sm_70   -DENABLE_CLANG_CUDA=Off     -DENABLE_HIP=Off   -DENABLE_TARGET_OPENMP=Off     -DCMAKE_CXX_COMPILER=g++     -DCMAKE_CXX_FLAGS="-Ofast -DNDEBUG" -DCMAKE_C_COMPILER=gcc -DCUDA_HOST_COMPILER=`which gcc`     ../

$ gcc --version
gcc (GCC) 8.1.0 20180502 (Cray Inc.)
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2019 NVIDIA Corporation
Built on Wed_Oct_23_19:24:38_PDT_2019
Cuda compilation tools, release 10.2, V10.2.89
```

(This is on the redwood test system, and I can help reproduce there if needed)

**Comment by abbotts** [comment]
Note: to reproduce this I needed to cherry-pick a WAR to issue #85

**Comment by rhornung67** [comment]
@abbotts It's taken me a while to work down my to-do list to get to this.  I've fixed the issue. A new Perf Suite release will appear in a few days and will contain the fix.  Thanks for pointing this out. 👍

**Comment by rhornung67** [comment]
Will be fixed in v0.8.0 release.
